libbeat/docs/upgrading.asciidoc

```diff
@@ -1,160 +1,324 @@
+----
+metricbeat setup --dashboards
+----
+
+Also, if you plan to run {beats} 6.7 and 7.0 in parallel,
+<<enable-ecs-compatibility,enable the compatibility layer>> *before* you start
+ingesting data. 
+
+[[upgrade-logstash-config]]
+==== Configure {ls} to use versioned templates and indices
+
+If you're using {ls} to ingest data, make sure {ls} uses versioned templates and
+indices. Otherwise, after you upgrade, {ls} will stop ingesting data because of
+field conflicts. To use versioned index templates and indices, configure the
+{ls} `index` setting as recommended in the
+{logstash-ref}/plugins-inputs-beats.html[{beats} input plugin] documentation and
+allow {beats} to manage the index template:
+
+[source,json]
+----
+    manage_template => false
+    index => "%{[@metadata][beat]}-%{[@metadata][version]}-%{+YYYY.MM.dd}" 
+----
+
+When you use this configuration, the index name is set to match the index
+pattern in the {beats} index template.
+
+IMPORTANT: The index name must match the index pattern in the {beats} index
+template, or {beats} dashboards and visualizations will not work as expected.
+For example, if {ls} sends events to an index called +metricbeat-7-{localdate}+,
+but the index template expects indices to match +metricbeat-{stack-version}-*+,
+your {beats} dashboards and visualizations will not display the data sent by
+{ls}.
+
+// REVIEWERS: The review comment said "things will break in a very opaque
+// manner." I'm assuming we mean that the Beats dashboards will stop working,
+// but I haven't had time to test this. ^^
+
+[[load-index-template-manually]]
+==== Load the 7.0 index template manually for outputs other than {es}
+
+If your {beats} configuration use an output other than `elasticsearch`, load
+the index template manually into {es} *before* ingesting any data into 7.0
+indices.
+
+To load the template manually, run the `setup` command with the `--template`
+flag specified. Temporarily enable the `elasticsearch` output when you run this
+command. For example:
 
-The old `-configtest` flag is still available, but deprecated.
+[source,shell]
+--
+metricbeat setup --template -E output.logstash.enabled=false -E 'output.elasticsearch.hosts=["localhost:9200"]'
+--
 
-===== Reference configuration files
+[[migrate-filebeat-registry]]
+==== Migrate {filebeat} registry to use new format
 
-The `<beatname>.full.yml` file, which contains all the non-deprecated
-configuration options is renamed to `<beatname>.reference.yml` starting with
-Beats 6.0. We recommend using this file as a reference only. It's not intended
-to be used in production.
+Starting with version 7.0, {filebeat} stores the registry in a directory
+structure. If Filebeat finds an old registry file at the path set by
+`filebeat.registry.path`, it will automatically migrate the registry file to the
+new format. If the registry file is at a different location, set
+`filebeat.registry.migrate_file` to point to the file.
```

Comment:
The registry changes also require users to change their settings. One needs to rename `filebeat.registry_file` to `filebeat.registry.path`. The other registry settings also require a dot with 7.0 (e.g. `registry_flush`, `registry.flush`).